components/ThemeSwitcher.vue

```diff
@@ -1,39 +1,25 @@
 <template>
-  <div class="mt-1">
-    <button
-      class="mr-1 h-min rounded-full bg-fog px-3 py-2 hover:bg-smoke dark:hidden sm:mx-2 md:px-4 md:py-3"
-      aria-roledescription="Dark Mode Toggle"
-      @click="handleClick"
-    >
-      <Icon
-        name="heroicons:moon"
-        class="hidden text-center text-lg dark:visible"
-      />
-    </button>
-    <button
-      class="mr-1 hidden h-min rounded-full bg-basalt px-3 py-2 text-white hover:bg-granite dark:block sm:mx-2 md:px-4 md:py-3"
-      aria-roledescription="Light Mode Toggle"
-      @click="handleClick"
-    >
-      <Icon
-        name="heroicons:sun"
-        class="hidden text-center text-lg dark:visible"
-      />
-    </button>
-  </div>
+  <button
+    class="flex h-10 w-10 items-center justify-center rounded-full"
+    :class="{
+      'bg-fog hover:bg-smoke': theme === 'light',
+      'bg-basalt text-white hover:bg-granite': theme === 'dark',
+      hidden: !theme,
+    }"
+    :aria-roledescription="
+      theme === 'light' ? 'Dark Mode Toggle' : 'Light Mode Toggle'
+    "
+    @click="toggleTheme"
+  >
+    <Icon
+      :name="theme === 'light' ? 'heroicons:sun' : 'heroicons:moon'"
+      class="text-center text-lg"
+    />
+  </button>
 </template>
 
-<script setup>
-useThemeHead();
+<script setup lang="ts">
+import { useThemeSwitcher } from '~/composables/useThemeSwitcher';
 
-const handleClick = () => {
-  const body = document.documentElement;
-  if (body.classList.contains('dark')) {
-    body.classList.remove('dark');
-    localStorage.setItem('theme', 'light');
-  } else {
-    body.classList.add('dark');
-    localStorage.setItem('theme', 'dark');
-  }
-};
+const { theme, toggleTheme } = useThemeSwitcher();
```

Comment:
Moved the theme management code into a composable in case we want to set the theme from other components